cmd/kops/delete.go

```diff
@@ -17,17 +17,55 @@ limitations under the License.
 package main
 
 import (
+	"fmt"
+	"os"
+	"strings"
+
+	"k8s.io/kops/util/pkg/ui"
+
 	"github.com/spf13/cobra"
 )
 
+var confirmDelete bool
+
 // deleteCmd represents the delete command
 var deleteCmd = &cobra.Command{
 	Use:        "delete",
 	Short:      "delete clusters",
 	Long:       `Delete clusters`,
 	SuggestFor: []string{"rm"},
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		// cobra doesn't give you the full arg list even though it should.
+		args = os.Args
+
+		// args should be [delete, resource, resource name]
+		// if there are less args than 3 confirming isnt necessary as the child command will fail
+		if !confirmDelete && len(args) >= 3 {
+			message := fmt.Sprintf(
+				"Do you really want to %s? This action cannot be undone.",
+				strings.Join(args, " "),
+			)
+
+			c := &ui.ConfirmArgs{
+				Out:     os.Stdout,
+				Message: message,
+				Default: "no",
+				Retries: 2,
```

Comment:
We might want to turn this up in the future, but I guess this is a good starting point